src/cipher.rs

```diff
@@ -509,102 +511,296 @@ impl rustls_root_cert_store {
     }
 }
 
+/// A builder for a `rustls_allow_any_authenticated_client_verifier`. This builder object can be
+/// used to configure certificate revocation lists, and then turned into a
+/// `rustls_allow_any_authenticated_client_verifier` once ready.
+pub struct rustls_allow_any_authenticated_client_builder {
+    _private: [u8; 0],
+}
+
+impl CastPtr for rustls_allow_any_authenticated_client_builder {
+    // XXX: contained value is consumed even on error, so this can contain None. but the caller
+    // still needs to free it
+    type RustType = Option<AllowAnyAuthenticatedClient>;
+}
+
+impl BoxCastPtr for rustls_allow_any_authenticated_client_builder {}
+
+impl rustls_allow_any_authenticated_client_builder {
+    /// Create a new allow any authenticated client certificate verifier builder using the root store.
+    ///
+    /// This copies the contents of the rustls_root_cert_store. It does not take
+    /// ownership of the pointed-to memory.
+    ///
+    /// This object can then be used to load any CRLs.
+    ///
+    /// Once that is complete, convert it into a real `rustls_allow_any_authenticated_client_verifier`
+    /// by calling `rustls_allow_any_authenticated_client_verifier_new()`.
+    #[no_mangle]
+    pub extern "C" fn rustls_allow_any_authenticated_client_builder_new(
+        store: *const rustls_root_cert_store,
+    ) -> *mut rustls_allow_any_authenticated_client_builder {
+        ffi_panic_boundary! {
+            let store: &RootCertStore = try_ref_from_ptr!(store);
+            let client_cert_verifier = Some(AllowAnyAuthenticatedClient::new(store.clone()));
+            BoxCastPtr::to_mut_ptr(client_cert_verifier)
+        }
+    }
+
+    /// Add one or more certificate revocation lists (CRLs) to the client certificate verifier by
+    /// reading the CRL content from the provided buffer of PEM encoded content.
+    ///
+    /// This function may return an error if the provided buffer is not valid PEM encoded content,
```

Comment:
```suggestion
    /// This function returns an error if the provided buffer is not valid PEM encoded content,
```

Super nitpicky but I think it's better to guarantee it _will_ return an error. :D

Reply:
Can't argue with that :-D Fixed.